samples/snippets/src/main/java/CustomCredentialSupplierOktaWorkload.java

```diff
@@ -0,0 +1,237 @@
+        oktaClientSecret);
+  }
+
+  public static void customCredentialSupplierOktaWorkload(
+      String gcpWorkloadAudience,
+      String serviceAccountImpersonationUrl,
+      String gcsBucketName,
+      String oktaDomain,
+      String oktaClientId,
+      String oktaClientSecret) {
+    // 1. Instantiate our custom supplier with Okta credentials.
+    OktaClientCredentialsSupplier oktaSupplier =
+        new OktaClientCredentialsSupplier(oktaDomain, oktaClientId, oktaClientSecret);
+
+    // 2. Instantiate an IdentityPoolCredentials with the required configuration.
+    IdentityPoolCredentials.Builder credentialsBuilder =
+        IdentityPoolCredentials.newBuilder()
+            .setAudience(gcpWorkloadAudience)
+            // This token type indicates that the subject token is a JSON Web Token (JWT).
+            // This is required for Workload Identity Federation with an OIDC provider like Okta.
+            .setSubjectTokenType("urn:ietf:params:oauth:token-type:jwt")
+            .setTokenUrl("https://sts.googleapis.com/v1/token")
+            .setSubjectTokenSupplier(oktaSupplier);
+
+    if (serviceAccountImpersonationUrl != null) {
+      credentialsBuilder.setServiceAccountImpersonationUrl(serviceAccountImpersonationUrl);
+    }
+
+    GoogleCredentials credentials = credentialsBuilder.build();
+
+    // 3. Use the credentials to make an authenticated request.
+    Storage storage = StorageOptions.newBuilder().setCredentials(credentials).build().getService();
+
+    System.out.println("[Test] Getting metadata for bucket: " + gcsBucketName + "...");
+    Bucket bucket = storage.get(gcsBucketName);
+    System.out.println(" --- SUCCESS! ---");
+    System.out.println("Successfully authenticated and retrieved bucket data:");
+    System.out.println(bucket.toString());
+  }
+
+  /**
+   * A custom SubjectTokenSupplier that authenticates with Okta using the Client Credentials grant
+   * flow.
+   */
+  private static class OktaClientCredentialsSupplier implements IdentityPoolSubjectTokenSupplier {
+
+    private final String oktaTokenUrl;
+    private final String clientId;
+    private final String clientSecret;
+    private String accessToken;
+    private long expiryTime;
+
+    public OktaClientCredentialsSupplier(String domain, String clientId, String clientSecret) {
+      this.oktaTokenUrl = domain + "/oauth2/default/v1/token";
+      this.clientId = clientId;
+      this.clientSecret = clientSecret;
+      System.out.println("OktaClientCredentialsSupplier initialized.");
+    }
+
+    /**
+     * Main method called by the auth library. It will fetch a new token if one is not already
+     * cached.
+     */
+    @Override
+    public String getSubjectToken(ExternalAccountSupplierContext context) throws IOException {
+      // Check if the current token is still valid (with a 60-second buffer).
+      boolean isTokenValid =
+          this.accessToken != null && System.currentTimeMillis() < this.expiryTime - 60 * 1000;
```

Comment:
expiryTime looks like it's set in `fetchOktaAccessToken()`, which is called later

Reply:
I think the logic is fine. From what I see, expiryTime is only set inside `fetchOktaAccessToken()`, so it may not be set on this first invocation (and has a default value of 0).